Automattic/jetpack
# [RNMobile] Remove mobile link from Account Connection card

As part of updates related to the Jetpack app's branding, we've decided to remove the mobile link from the Account Connection card found via _Jetpack_ → _Dashboard_. 

#### Changes proposed in this Pull Request:

* The `MobileMagicLink` component has been removed from the Account Connection card. 
* The code that was only used by [the `MobileMagicLink` component](https://github.com/Automattic/jetpack/tree/trunk/projects/plugins/jetpack/_inc/client/components/mobile-magic-link) has also been deleted following [the discussion here](https://github.com/Automattic/jetpack/pull/26311#pullrequestreview-1117418936). The component was originally introduced in https://github.com/Automattic/jetpack/pull/12561 and that PR can be referred to to confirm the correct code has been deleted.
* The removal of the now unused code includes a changes to `js-packages/api`, which means that `api/package.json` had to be bumped from version `0.13.11` to `0.14`. This also included changes to various dependencies, as shown in https://github.com/Automattic/jetpack/pull/26311/commits/2375e4c2a7203f8938159b27ffd6ec2a0d4144dc.

#### Other information:


#### Jetpack product discussion

pcdRpT-F0-p2

#### Does this pull request change what data or activity we track or use?

We'll no longer be tracking clicks on the mobile login link, as the `MobileMagicLink` component is no longer in use anywhere. 

#### Testing instructions:

* Navigate to _Jetpack_ → _Dashboard_ and verify that the link is no longer displayed in the Account Connection card.

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/191475730-705d5942-448c-49bf-af84-7d007bca7fff.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/191475211-0a8a653a-f189-4ef3-ac04-748163e4fd04.png" width="100%"> |
